### ISSUE TITLE
Custom auth handler closes connection without return

### ISSUE THREAD

**Issue by kica-z** [civetweb/civetweb#1142]
https://github.com/civetweb/civetweb/blob/d5c0f6d8460b6f92f8a7429c7856b08cca42a863/src/civetweb.c#L14968
Currently if the custom authenticator returns `false` the connection is just closed without any feedback to the client. Wouldn't it be better to return a `401` response in this case?
```
if (!auth_handler(conn, auth_callback_data)) {
    mg_send_http_error(conn,
                       401,
                       "%s Unauthorized",
                       conn->request_info.request_method);
    /* Callback handler will not be used anymore. Release it */
    release_handler_ref(conn, handler_info);

    return;
}
``` 
Or is there something i am not seeing at the Moment?

**Comment by bel2125** [comment]
The idea is, the handler could call this `mg_send_http_error(conn,
                       401,
                       "%s Unauthorized",
                       conn->request_info.request_method);
 ` itself. 
Or a 403 Forbidden:  https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
Or a 402 Payment Required: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402
Or a 303 redirect to a login page: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303

**Comment by bel2125** [comment]
Not sending anything by default allows the implementation of the callback to send whatever appears the most suitable in this case. In many cases this would be a 401. But it could be something else as well, then it is better the server does not implicitly assume anything.

**Comment by kica-z** [comment]
Agreed, this sounds plausible and like the most flexible solution. Since normally(mostly) `auth_handlers` should not have side effects and this behaviour is (at least for me) not quite intuitive I would at least mention this in the documentation. I can make a PR if you want. Where would this have to be documented? Here?
https://github.com/civetweb/civetweb/blob/master/docs/api/mg_set_auth_handler.md
